architectures/BDR-Always-ON/commands/upgrade.yml

```diff
@@ -98,6 +103,22 @@
   # (e.g., primary instances) or not (e.g., logical standby instances).
   # We don't do anything for non-Postgres instances at the moment.
 
+  - name: Check to perform major upgrade for Postgres
+    set_fact:
+      postgres_major_version_upgrade: "{{ cluster_facts.postgres_version is version(postgres_version, 'lt') }}"
```

Comment:
How does this get around the `Ensure postgres_version matches pgdata_version, if defined`
 assert in init/tasks/postgres.yml?

Reply:
The comparison here is between `postgres_version` declared in `cluster_vars` and `postgres_version` declared under the vars of a host ready for upgrade. Actually it is this comparison that enables and drives the in-place upgrades. But I think to answer your question, the `pgdata_version` and `postgres_version` variables are relative to the host and should match by calling `postgres/service` after an upgrade.

To note, the variables `postgres_version` and `postgres_data_dir` are created as separate host vars in the updated config for hosts ready to upgrade. The updated config is then provisioned before running `upgrade` for those hosts. After running `upgrade`, deploy tidies everything up.

Reply:
@amenonsen I think I could use cluster_vars instead of cluster_facts here and probably some other places where it makes more sense given the change in cluster_facts as the upgrade modifies the actual cluster properties or facts on individual nodes.